illumos-utils/src/zfs.rs

```diff
@@ -363,7 +380,8 @@ impl Zfs {
         name: &str,
         mountpoint: &Mountpoint,
         quota: Option<usize>,
-        compression: Option<&'static str>,
+        reservation: Option<usize>,
```

Comment:
Ha, I made this same comment in my review. I think we might want a way to explicitly distinguish three things: "set this reservation to this value", "do not change the reservation", and "unset the reservation". Ditto for quota and probably compression too.

Reply:
I updated this in c5621256ccc7a8995842d057e608c8b3b66effe5 -- I'm going to use a policy of `None` implies "set as none/off", because I think this is in-line with how the reconfigurator will want to set state here. Namely: once we integrate datasets in the blueprint, we want to "ensure that they exist, and conform them to a specification" which matches how we assign zones to sleds.